src/app/_components/_sharedcomponents/Cards/GameCard/GameCard.tsx

```diff
@@ -2,51 +2,112 @@ import React from "react";
 import {
 	Card as MuiCard,
 	CardContent,
-	CardActionArea,
 	Typography,
 	Box,
 } from "@mui/material";
 import Image from "next/image";
-import { IGameCardProps, ICardData } from "@/app/_components/_sharedcomponents/Cards/CardTypes";
+import { IGameCardProps, ICardData, IServerCardData } from "@/app/_components/_sharedcomponents/Cards/CardTypes";
 import { useGame } from "@/app/_contexts/Game.context";
 import { s3CardImageURL } from "@/app/_utils/s3Utils";
 
+// Type guard to check if the card is ICardData
+const isICardData = (card: ICardData | IServerCardData): card is ICardData => {
+	return (card as ICardData).uuid !== undefined;
+};
+
 const GameCard: React.FC<IGameCardProps> = ({
 	card,
 	size = "standard",
+	location = "gameBoard",
+	pile,
+	options,
 }) => {
 	// const isLobbyView = path === "/lobby";
 	const isLobbyView = false;
-	const isFaceUp = true
-
-	const { sendGameMessage } = useGame();
-
+	const isFaceUp = true;
+	// Determine whether card is ICardData or IServerCardData
+	const cardData = isICardData(card) ? card : card.card;
+	const cardCounter = !isICardData(card) ? card.count : 0;
+	const { sendGameMessage, connectedDeck, setConnectedDeck } = useGame();
 	const cardBorderColor = (card: ICardData) => {
 		if (card.selected) return "yellow";
 		if (card.selectable) return "limegreen";
 		if (card.exhausted) return "gray";
 		return "";
 	}
 
+	// handle deck movement
+	const handleSwitch = () => {
```

Comment:
This is Lobby specific logic that would be better suited living in the Lobby Deck component. In general this component should not care about what page or pile it belongs to. We wnat to tell the component how we would like it to render in those components and pass those options as props. If we want to change the onclick behavior of the component we could pass that as handler through the props. For the game card I think since in most cases this will be on the gameboard it makes sense to have the game message as it's default handler and then use the onclick handler provided in props as an override if one is provided.